File: scanner/corpora/python/positive/aes_gcm.py
"""AES-GCM encryption and decryption using PyCryptodome."""
from Crypto.Cipher import AES
from Crypto.Random import get_random_bytes

key = get_random_bytes(32)  # 256-bit key
cipher = AES.new(key, AES.MODE_GCM)
nonce = cipher.nonce
plaintext = b"Secret message for GCM mode"
ciphertext, tag = cipher.encrypt_and_digest(plaintext)

# Decrypt
cipher_dec = AES.new(key, AES.MODE_GCM, nonce=nonce)
decrypted = cipher_dec.decrypt_and_verify(ciphertext, tag)
print(decrypted)

File: scanner/corpora/go/positive/aes_gcm.go
package main

import (
	"crypto/aes"
	"crypto/cipher"
	"crypto/rand"
	"crypto/sha256"
	"io"
	"fmt"
)

func aesGcmEncrypt(plaintext, key []byte) ([]byte, []byte, error) {
	block, err := aes.NewCipher(key)
	if err != nil {
		return nil, nil, err
	}
	aesGCM, err := cipher.NewGCM(block)
	if err != nil {
		return nil, nil, err
	}
	nonce := make([]byte, aesGCM.NonceSize())
	if _, err := io.ReadFull(rand.Reader, nonce); err != nil {
		return nil, nil, err
	}
	ciphertext := aesGCM.Seal(nonce, nonce, plaintext, nil)
	return ciphertext, nil, nil
}

func main() {
	key := make([]byte, 32)
	rand.Read(key)
	data := []byte("Secret GCM message")
	ct, _, err := aesGcmEncrypt(data, key)
	if err != nil {
		panic(err)
	}
	// Compute hash
	h := sha256.Sum256(ct)
	fmt.Printf("Hash: %x\n", h)
}
